drivers/SmartThings/zigbee-fan/profiles/kichler-fan-light.yml

```diff
@@ -0,0 +1,67 @@
+name: kichler-fan-light
+components:
+  - id: main
+    label: Fan 
+    capabilities:
+      - id: switch
+        version: 1
+      - id: fanSpeed
+        version: 1
+        config:
+          values:
+            - key: "fanSpeed.value"
+              range: [0, 3]
+      - id: firmwareUpdate
+        version: 1
+      - id: refresh
+        version: 1
+    categories:
+      - name: Fan
+  - id: light
+    label: Light
+    capabilities:
+      - id: switch
+        version: 1
+      - id: switchLevel
+        version: 1
+        config:
+          values:
+            - key: "level.value"
+              range: [0, 100]
+      - id: refresh
+        version: 1
+    categories:
+      - name: Light
+preferences:
+  - name: "trim"
+    title: "Minimum Dimming Level"
+    description: "Set the lowest dimming level"
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        10 : "10%"
+        15 : "15%"
+        20 : "20%"
+        25 : "25%"
+      default: 10
```

Comment:
In relation to this preference, I'm happy to direct you to this new feature we added, which is the ability to set min/max values for the switch level attribute. You may find that using this is simpler than having to do the trimming yourself.

Here's an example of how to emit this event:

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1220/files#diff-7ab55ddbc6a9760ea5b6f928abee244c3142c74389743ce05b895b938ad81f10R372

The min/max _should_ be displayed and carried across automations as well.

Reply:
It is expected that there will be models derived from this model. We will try to use the code you suggested at that time. Thank you.